code/handlers/trackservers.q

```diff
@@ -175,12 +173,15 @@ retrydiscovery:{
 // Called by the discovery service when it restarts
 autodiscovery:{if[DISCOVERYRETRY>0; .servers.retrydiscovery[]]}
 
+//check if .proc.getattributes exists in the nontorqprocess
+checknontorqattr:{$[(`proc in key`)&(first distinct`getattributes`cd in key`.proc);1;0]};
```

Comment:
I don't get the second condition here ``first distinct`getattributes`cd in key`.proc``, won't `first` return the same as `first distinct`? And if you're running `first` then why not just check for `` `getattributes`` only as you'll never get to `` `cd``. Also why is this a conditional? Should just return from the condition directly unless it needs to be a long.

Reply:
even if it does need to be a long, could just cast it & still remove conditional

Reply:
yeah. I have seen now that it basically returns the same thing. I will remove the conditional form as well.